test/test_main_invocations.py

```diff
@@ -0,0 +1,136 @@
+# Copyright 2024 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
```

Comment:
This test module feels a little different than the usual ones - it does test that calling main does/doesn't raise an exception in various ways, but it's also rely on the type checker being run against this file and finding any issues, which is why there are (e.g.) fewer asserts than you might expect.

If I'm understanding that correctly, could we have a docstring that briefly explains that? I think that would make it easier for people later on to understand what's happening here and why things are the way they are.

Reply:
Sure we can :)

Btw., if you folks prefer, I can separate run-time tests and typing tests.

WDYT?

Reply:
I've added the test module docstring for now.

Reply:
Would you rather prefer this, or https://github.com/canonical/operator/pull/1339 ?

Reply:
I've split up the tests.